### PR TITLE
Issue #9142: resolve Assertions.assertEquals

### PR DESCRIPTION
**Pull request by rnveach**
Issue #9142

Code was automatically replaced with https://github.com/rnveach/checkstyle/commits/truth_replacer . I just went back and fixed the checkstyle errors and formatting.

### CODE REVIEW

**Comment by rnveach** [comment]
Wercker failure is unrelated.

**Review by romani — APPROVED** [review]
Enormous wor, I don't scroll up to the end.

**Comment by rnveach** [comment]
This is the largest work of the entire resolve. Like I said above, replacements was automated with basically an auto-fix. Only some formatting had to be done manually.

File: src/test/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinterTest.java [hunk 1]
```diff
@@ -61,8 +60,9 @@ public void testParseFileThrowable() throws Exception {
         }
         catch (CheckstyleException ex) {
             assertSame(IllegalStateException.class, ex.getCause().getClass(), "Invalid class");
-            assertEquals("2:0: no viable alternative at input 'classD'",
-                    ex.getCause().getMessage(), "Invalid exception message");
+            assertWithMessage("Invalid exception message")
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinterTest.java:63]: automatically replace 
```java
.that(ex.getCause().getMessage())
                .isEqualTo("str");
```
with more truth style
```java
.that(ex)
 .hasCauseThat()
    .hasMessageThat()
    .isEqualTo("str");
```
?

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 2]
```diff
@@ -667,11 +693,13 @@ public void testClearExistingCache() throws Exception {
...
-        assertEquals(cacheAfterClear.getProperty(PropertyCacheFile.CONFIG_HASH_KEY),
-                cacheHash, "Cash have changed it hash");
+        assertWithMessage("Cash have changed it hash")
+            .that(cacheHash)
+            .isEqualTo(cacheAfterClear.getProperty(PropertyCacheFile.CONFIG_HASH_KEY));
         final int expectedNumberOfObjectsInCacheAfterSecondRun = 2;
-        assertEquals(expectedNumberOfObjectsInCacheAfterSecondRun, cacheAfterSecondRun.size(),
-                "Cache has changed number of items");
+        assertWithMessage("Cache has changed number of items")
+            .that(cacheAfterSecondRun.size())
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:701]: and this
```java
.that(cacheAfterSecondRun.size())
.isEqualTo(expectedNumberOfObjectsInCacheAfterSecondRun);
```
to 
```java
.that(cacheAfterSecondRun)
.hasSize(expectedNumberOfObjectsInCacheAfterSecondRun);
```

**Review by pbludov — APPROVED** [review]
How difficult it will be to extend the truth_replacer to make it more Truth-style?
See comments below:

**Comment by rnveach** [comment]
@pbludov Can we move these replacements to a new issue and start a full list of what needs to replaced to what else.

One issue with the replacement is it does not help with formatting, so it requires me to go back and format the pieces and as you saw here, it is a large list.

Another is `truth_replacer` is a very simple replacer. It only looks for the method name and replaces the entire method call strain plopping individual parameters into individual locations. The pieces you have provided need more indepth analysis on whats being passed to the method to determine if it can be replaced.

I will work on these if delayed after I complete this entire issue as I have like 5 more PRs set up to go.

**Comment by rnveach** [comment]
I also highly recommend if we want to keep this in the future, we look at creating a check to maintain it as this will go missed in future updates.

**Comment by pbludov** [comment]
OK, let's merge this PR in and move on.

**Comment by pbludov** [comment]
> I also highly recommend if we want to keep this in the future, we look at creating a check to maintain it as this will go missed in future updates. Anything not maintained by the CI will eventually be missed by maintainers.

To do this, we need to know what type the methods `size`, `endsWith`, etc. belong to. In order to avoid false positives.
This can be done with a StopBugs detector.

**Comment by romani** [comment]
I just fixed all cases .... but you merged them too quickly ....

**Comment by pbludov** [comment]
> I just fixed all cases .... but you merged them too quickly ....

My bad. Please revert my commit.

**Pull request merged**